torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -766,6 +766,14 @@ void enableProfiler(
 }
 
 std::unique_ptr<ProfilerResult> disableProfiler() {
+  // In PyTorch Lightning, it would call disableProfiler first before enabling it
```

Comment:
Do I understand properly that the problem here is that this function should only be called if the profiler is enabled. But you want it to be a no-op in such case?

Reply:
yes and @robieta already point me that there is already one flag to check whether the profiler is enabled or not